packages/webapp/src/elections/components/ongoing-election-components/ongoing-round-segment.tsx

```diff
@@ -0,0 +1,257 @@
+    const [selectedMember, setSelected] = useState<MemberData | null>(null);
+    const [
+        showZoomLinkPermutations,
+        setShowZoomLinkPermutations,
+    ] = useState<boolean>(false); // TODO: Replace with real meeting link functionality
+    const [showPasswordPrompt, setShowPasswordPrompt] = useState<boolean>(
+        false
+    ); // TODO: Hook up to the real password prompt
+
+    const { data: loggedInMember } = useCurrentMember();
+
+    const { data: voteData } = useMemberGroupParticipants(
+        loggedInMember?.account
+    );
+
+    useEffect(() => {
+        if (voteData) setVoterStats(voteData);
+    }, [voteData]);
+
+    const roundEdenMembers = useMemberListByAccountNames(
+        voteData?.map((participant) => participant.member) ?? []
+    );
+
+    const nftTemplateIds: number[] = roundEdenMembers
+        ?.filter((res) => Boolean(res?.data?.nft_template_id))
+        ?.map((res) => res?.data as EdenMember)
+        .map((em) => em.nft_template_id);
+
+    useEffect(() => {
+        const roundMemberError = roundEdenMembers.some((res) => res.isError);
+        if (roundMemberError) {
+            setFetchError(true);
+        } else {
+            setFetchError(false);
+        }
+    }, [roundEdenMembers]);
+
+    const { data: members } = useQuery({
+        ...queryMembers(1, 20, nftTemplateIds),
+        staleTime: Infinity,
+        enabled: !fetchError && Boolean(nftTemplateIds.length),
+    });
+
+    // TODO: Handle Fetch Errors;
+    if (fetchError) return <Text>Fetch Error</Text>;
+    if (!members || members?.length !== voteData?.length)
+        return <Text>Error Fetching Members</Text>;
+
+    const onSelectMember = (member: MemberData) => {
+        if (member.account === selectedMember?.account) return;
+        setSelected(member);
+    };
+
+    const userVoterStats = voterStats.find(
+        (vs) => vs.member === loggedInMember?.account
+    );
+
+    const userVotingFor = members.find(
+        (m) => m.account === userVoterStats?.candidate
+    );
+
+    // TODO: Sign & push vote action; remove voterStats and switch back to relying on voteData
+    const onSubmitVote = () => {
+        setVoterStats([
+            ...voterStats.filter((vs) => vs.member !== loggedInMember?.account),
+            {
+                ...userVoterStats!,
+                candidate: selectedMember?.account ?? "",
+            },
+        ]);
+    };
+
+    // TODO: If we want the list leaderboard flipper animation, we'll want to poll with the query and sort round participants by number of votes.
+    // Then we'll feed that into the Flipper instance below.
+    const getVoteCountForMember = (member: MemberData) => {
+        return voterStats.filter((vd) => vd.candidate === member.account)
+            .length;
+    };
+
+    const sortMembersByVotes = [...members].sort(
+        (a, b) => getVoteCountForMember(b) - getVoteCountForMember(a)
+    );
```

Comment:
I'm reading about how best to do it, but there's likely a simple encapsulation to offload a lot of this init code to the bottom of the file or a separate module, but I'm sure that's part of your stated refactor. So I'll leave this be for now.

Reply:
Yeah, there's a lot of it, isn't there! But yes, once we have it working with the real data, we'll encapsulate this. Maybe a hook, maybe otherwise.